controllers/config/config.go

```diff
@@ -68,12 +68,94 @@ func fillConfig(clusterConfig *configv1.Network, operConfig *operatorv1.AntreaIn
 	return nil
 }
 
+func fillControllerConfig(clusterConfig *configv1.Network, operConfig *operatorv1.AntreaInstall) error {
+	var controllerConfig ctlconfig.ControllerConfig
+	err := yaml.Unmarshal([]byte(operConfig.Spec.AntreaControllerConfig), &controllerConfig)
+	if err != nil {
+		return fmt.Errorf("failed to parse AntreaControllerConfig: %v", err)
+	}
+
+	//Turn on NodeIPAM featureGate.
+	if controllerConfig.FeatureGates == nil {
+		controllerConfig.FeatureGates = make(map[string]bool)
+	}
+	controllerConfig.FeatureGates[string(features.NodeIPAM)] = true
+	controllerConfig.NodeIPAM.EnableNodeIPAM = true
+
+	if clusterConfig != nil {
+		ip4found := false
+		ip6found := false
+
+		for _, cidr := range clusterConfig.Spec.ClusterNetwork {
+			if net.IsIPv4CIDRString(cidr.CIDR) {
+				if !ip4found {
+					ip4found = true
+					controllerConfig.NodeIPAM.ClusterCIDRs = append(controllerConfig.NodeIPAM.ClusterCIDRs, cidr.CIDR)
```

Comment:
I am not sure I understand the logic around the `ip4found` and `ip6found` variables. Can you explain?

Reply:
clusterConfig.Spec.ClusterNetwork contains a list of IP4, IP6 CIDRs.
I would like to take the first IP4 and first IP6 CIDRs from the list and append these into the controller config.
I could basically assign the while list from the cluster to the controller config but I need their matching prefix lengths as well.

Reply:
Thanks for the clarification. Will this code eventually use all the CIDRs in `clusterConfig.Spec.ClusterNetwork`... I think we would like to make sure all of them are used

Reply:
Not sure whether this is a limitation applied by Antrea Agent or by NodeIPAM itself. So it doesn't matter what this code does but how Antrea or NodeIPAM utilizes these CIDRs.